plugins/redis/secret_key.go

```diff
@@ -0,0 +1,55 @@
+package redis
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func SecretKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.SecretKey,
+		DocsURL: sdk.URL("https://docs.redis.com/latest/rc/api/get-started/manage-api-keys/"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.AccessKey,
+				MarkdownDescription: "API account key (also known as Access Key, or just API Key) to authenticate to Redis Enterprise Cloud.",
+				Secret:              true,
+				Optional:            false,
+				Composition: &schema.ValueComposition{
+					Length: 51,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.SecretKey,
+				MarkdownDescription: "API user key (also known as Secret Key) to authenticate to Redis Enterprise Cloud.",
+				Secret:              true,
+				Optional:            false,
+				Composition: &schema.ValueComposition{
+					Length: 50,
```

Comment:
Hmm. I am getting a key of length 51 when generating a User API Key. That makes me wonder whether this length is even fixed at all. I'd consider not setting this value to avoid over fitting this rule.